### PR TITLE
GroupData: WebXR Device API to latest spec

### PR DESCRIPTION
**Pull request by a2sheppy**
This PR makes the changes needed to bring the
WebXR Device API spec up to the current state of
the specification. Also moves some items which
were incorrectly listed as "types" into the
"dictionaries" section for WebXR. I've opted to
combine all the WebXR specs in GroupData since
they are all so closely bound together.

Sources:
* https://immersive-web.github.io/webxr
* https://immersive-web.github.io/webxr-ar-module
* https://immersive-web.github.io/webxr-gamepads-module

### CODE REVIEW

File: macros/GroupData.json [hunk 1]
```diff
@@ -1754,31 +1754,30 @@
                             "XRViewerPose",
                             "XRViewport",
                             "XRWebGLLayer" ],
-            "methods":    [ ],
+            "methods":    [ "WebGLRenderingContextBase.makeXRCompatible()" ],
```
Comment (Elchi3) [inline macros/GroupData.json]: "WebGLRenderingContextBase" isn't a thing that is exposed to web developers, so we didn't document it this way. I think you want WebGLRenderingContext here.

Comment (a2sheppy) [reply macros/GroupData.json]: Yes, you're right. Will fix.

Comment (jpmedley) [reply macros/GroupData.json]: @Elchi3 The suggestion seems right. Is this the thing you wanted me to look at?

Comment (Elchi3) [reply macros/GroupData.json]: This and possible the other changes here, since I'm not a subject matter expert.

**Comment by jpmedley** [comment]
Everything in this PR looks right to me.

**Pull request merged**